connectors/services/job_scheduling.py

```diff
@@ -88,9 +88,19 @@ async def _schedule(self, connector):
             raise DataSourceError(msg)
 
         source_klass = get_source_klass(self.source_list[connector.service_type])
+        data_source = source_klass(connector.configuration)
+        data_source.set_logger(connector.logger)
+
+        connector.log_debug("Validating configuration")
+        try:
+            data_source.validate_config_fields()
```

Comment:
Isn't this validated by `prepare`?
https://github.com/elastic/connectors/blob/17a76e99c2b3fc0151ba5c65eceb541f84dc6b2f/connectors/services/job_scheduling.py#L51

Reply:
I haven't found any validation logic in prepare.

`connector.prepare` has no access to instance of data source with configuration, rather it's a wrapper on top of an index record that does limited "sanity" validation for the connector fields, not RCFs.

Reply:
The `data_source.validate_config_fields()` here validates whether there are missing configs in the connector's configuration. 

https://github.com/elastic/connectors/blob/8cae11f4d9e652c047a5a970e599e066645f29e7/connectors/source.py#L546-L555

In `prepare`, it tries to call `validated_doc`, which will simply update the connector's configs if there's missing configs

https://github.com/elastic/connectors/blob/8cae11f4d9e652c047a5a970e599e066645f29e7/connectors/protocol/connectors.py#L807

Reply:
Hmmmm, I copied this logic from another place that validates fields:

https://github.com/elastic/connectors/blob/main/connectors/sync_job_runner.py#L129-131

```python
            self.sync_job.log_debug("Validating configuration")
            self.data_provider.validate_config_fields()
            await self.data_provider.validate_config()
```

Should we remove it in both places?

I'd honestly keep it, cause we haven't fully addressed migrating fields back and forth (deleting/modifying), so it _could_ work as some fail-safe mechanism that will raise if we migrated something wrong. How do you feel?

Reply:
When a sync job is created, it copies the configuration from the connector, and when it's executed, it will use the configuration in the sync job.

The validation in sync_job_runner is to validate the config in the sync job and should be kept.

In Job scheduling, it tries to check for each connector whether there's any discranpancy between the connector's configuration and the data source's configuration, if so, it will try to update the connector's, instead of raising an error. This is happening in the `prepare` method.

I doubt that the validation you added will ever raise, because it happens after `prepare`.